fix(types) Export cbs and their parameter types

dougalg opened the pull request:
<!-- Please fill in the following information to help us review your PR more efficiently. -->

## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others



### What does this PR do?

<!-- USE ONCE SENTENCE TO DESCRIBE WHAT THIS PR DOES. -->

Adds type exports for several callbacks and their parameters

### Fixed issues

<!--
- #xxxx: ...
-->

- #14277: Export types for formatter callback params

## Details

### Before: What was the problem?

<!-- DESCRIBE THE BUG OR REQUIREMENT HERE. -->

<!-- ADD SCREENSHOT HERE IF APPLICABLE. -->

It was difficult to implement properly types callback functions

### After: How is it fixed in this PR?

<!-- THE RESULT AFTER FIXING AND A SIMPLE EXPLANATION ABOUT HOW IT IS FIXED. -->

<!-- ADD SCREENSHOT HERE IF APPLICABLE. -->

It should now be possible to implement properly typed callbacks

## Misc

<!-- ADD RELATED ISSUE ID WHEN APPLICABLE -->

- [ ] The API has been changed (apache/echarts-doc#xxx).
- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

As far as I could tell, there are no tests needed for type exports, but if I missed them, please let me know, and I will try to add them.

## Others

### Merging options

- [ ] Please squash the commits into a single one when merge.

### Other information

pissang commented on src/export/callback.ts:
```diff
@@ -0,0 +1,25 @@
+import type {
```
Needs Apache License in the header.

dougalg replied:
Thanks for the feedback! I was considering putting them in there, but I initially wasn't sure how many exports I'd be adding and I didn't want to bloat that file too much. But I think this doesn't add too much bloat, although there are some more callbacks I didn't include as I wasn't sure exactly how they should be exported.

dougalg commented:
@pissang I have moved this PR to "awaiting review" from WIP, whenever you (or others) are ready to do a final/next review.

Thank you again!

pissang commented on src/util/types.ts:
```diff
@@ -1235,7 +1235,7 @@ type TooltipBoxLayoutOption = Pick<
  * Position relative to the hoverred element. Only available when trigger is item.
  */
-interface PositionCallback {
+export interface PositionCallback {
```
PositionCallback can be renamed to TooltipPositionCallback to avoid potential conflicts when exported

pissang commented on src/export/option.ts:
```diff
@@ -234,4 +250,17 @@ export interface EChartsOption extends ECBasicOption {
+    AnimationDelayCallbackParam as AnimationDelayCallbackParams,
+    LabelLayoutOptionCallbackParams,
+    LabelLayoutOptionCallback,
+    PositionCallback
```
`PositionCallback` is only used in the tooltip. So it can also be TooltipComponentPositionCallback

dougalg commented:
@pissang I believe I have addressed your comments. I also noticed some inconsistencies in my naming patterns and renamed a couple of the exports to be more consistent. Thank you very much
